xGame/Java/xgame/src/test/java/com/xgame/restservice/ChessMatchService.java

```diff
@@ -135,4 +135,34 @@ void acceptInvite() {
 			userRepo.delete(player2);
 		}
 	}
+	
+	@Test 
+	void getMatchById(){
+		var matchCount = matchRepo.count();
+		
+		var player1 = userRepo.save(new User("junit1", "junit1@email.com", "junit1password"));
+		var player2 = userRepo.save(new User("junit2", "junit2@email.com", "junit2password"));
+		
+		var testMatch = service.createMatch(player1.getId(), player2.getId());
+		var testMatchFromRepo = service.getMatch(testMatch.getId());
+		
+		try {
+			assertNotNull(testMatchFromRepo);
+			assertEquals(matchCount + 1, matchRepo.count());
+			assertEquals(testMatchFromRepo.getWhiteId(), player1.getId());
+			assertEquals(testMatchFromRepo.getBlackId(), player2.getId());
+			assertEquals(testMatchFromRepo.getWhiteEmail(), player1.getEmail());
+			assertEquals(testMatchFromRepo.getBlackEmail(), player2.getEmail());
+			assertEquals(testMatchFromRepo.getTurnCount(), 0);
+		}
+		catch(Exception e) {
+			fail();
+		}
+		finally {
+			//cleanup
+			matchRepo.deleteById(testMatchFromRepo.getId());
+			userRepo.delete(player1);
+			userRepo.delete(player2);
+		}
+	}
```

Comment:
Could you also add a test for when no match is found and an exception is thrown?

Reply:
I agree with Lance this is a good idea.

Reply:
Just added, let me know if you're looking for something different

Reply:
Make it something dumb like -1 or 0. Something that can never occur in the database organically.